src/base.jl

```diff
@@ -2098,13 +2152,27 @@ function handle_component_removal!(sys::System, device::Device)
     return
 end
 
+function handle_component_removal!(sys::System, component::RegulationDevice)
```

Comment:
We had two existing bugs where we were not removing masked components from RegulationDevice or StaticInjectionSubsystem.